src/libcore/iter/traits/accum.rs

```diff
@@ -119,67 +120,56 @@ float_sum_product! { f32 f64 }
 ///
 /// If an error is encountered, the iterator stops and the error is
 /// stored. The error may be recovered later via `reconstruct`.
-struct ResultShunt<I, E> {
+struct ResultShunt<'a, I, E> {
     iter: I,
-    error: Option<E>,
+    error: &'a mut Result<(), E>,
 }
 
-impl<I, T, E> ResultShunt<I, E>
-    where I: Iterator<Item = Result<T, E>>
+/// Process the given iterator as if it yielded a `T` instead of a
+/// `Result<T, _>`. Any errors will stop the inner iterator and
+/// the overall result will be an error.
+fn process_results<I, T, E, F, U>(iter: I, mut f: F) -> Result<U, E>
+    where I: Iterator<Item = Result<T, E>>,
+          for<'a> F: FnMut(ResultShunt<'a, I, E>) -> U
```

Comment:
I recommend running rustfmt on newly-touched code.

Reply:
Heh, I tried to match the style of the file but it seems to already be a mix anyway. I'll change it.